### PR TITLE
[PLAT-494] Optimize Cos.io

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

Cos.io currently runs cos.i-slow this fix addresses some of the things slowing it down.

## Changes

- Uses smaller image instead of massive 4000x2000 image on menu page
- Prefetch photos takes queries from about 150 to about 100

## Side Effects 

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/PLAT-494

### CODE REVIEW

File: common/models.py [hunk 1]
```diff
@@ -395,19 +395,19 @@ class CustomPage(Page, index.Indexed):
     def serve(self, request):
         return render(request, self.template, {
             'page': self,
-            'people': Person.objects.all(),
+            'people': Person.objects.all().prefetch_related('photo'),
```
Comment (sloria) [inline common/models.py]: Drive-by: I think you can use `select_related` rather than `prefetch_related` since `photo` is a many-to-one relationship. That will save even more queries.

File: common/models.py [hunk 2]
```diff
@@ -395,19 +395,19 @@ class CustomPage(Page, index.Indexed):
...
             'team': Person.objects.filter(
                 tags__name__in=['team']
-            ).order_by('last_name'),
+            ).prefetch_related('photo').order_by('last_name'),
             'alumni': Person.objects.filter(
                 tags__name__in=['alum']
-            ).order_by('last_name'),
+            ).prefetch_related('photo').order_by('last_name'),
```
Comment (sloria) [inline common/models.py]: Drive-by: since we don't show photos for alumni, there's no need to prefetch it.

In fact, you can use `.only` to load only the fields we need. I think we only need the first and last name fields.

File: common/models.py [hunk 3]
```diff
@@ -395,19 +395,19 @@ class CustomPage(Page, index.Indexed):
...
             'ambassadors': Person.objects.filter(
                 tags__name__in=['Ambassador']
-            ).order_by('last_name'),
+            ).prefetch_related('photo').order_by('last_name'),
             'topcoordinators': Person.objects.filter(
                 tags__name__in=['TOP-coordinator']
-            ).order_by('last_name'),
+            ).prefetch_related('photo').order_by('last_name'),
```
Comment (sloria) [inline common/models.py]: Same here: remove prefetching if it isn't necessary.

**Comment by erinspace** [comment]
You might consider trying to use django debug toolbar or n plus one while in dev mode like we do osfside as a part of this PR -- while you probably won't fix every problem these tools reveal, it'll at least give us a really good baseline for how to improve in the future. An attempt at that plus addressing steve's drivebys will be good for another CR pass!

**Comment by erinspace** [comment]
Just a question! Did you try checking out the debug toolbar to find more pages/queries to be optimized? How about nplusone? If you find ones that would take a significant amount of time to fix, you can create more specific tickets to fix them later, but I believe those should still be documented under the scope of this ticket.

**Comment by Johnetordoff** [comment]
@erinspace I did use the toolbar to find duplicate queries. It's pretty difficult to actually address these because most of the queries are for wagtail assets and not typical stuff, and even when I identify the assets which are causing duplicated queries I can't do much because they are baked into wagtail. 

This is a pretty broad ticket and I wish I had more experience with optimization so I could know for sure, but I think it's just the number of assets on the page and the way wagtail is loading them that are responsible for the slowness.

**Review by sloria — APPROVED** [review]
This should improve load times for the team page. LGTM
